packages/voter-dapp/src/RetrieveRewards.js

```diff
@@ -85,24 +85,32 @@ function useRetrieveRewardsTxn(retrievedRewardsEvents, reveals, votingAccount) {
       }
     }
 
+    // Check retrieveRewards return value before calling.
+    const VotingContract = drizzle.contracts.Voting;
+    const pendingRewardsPromise = VotingContract.methods
```

Comment:
why do you not use the `useCache` here?

Reply:
I wasn't able to get `useCacheCall` to work on the `retrieveRewards` function, and `useCacheSend` doesn't provide a way to just `.call()` a write-transaction. Source here: https://github.com/UMAprotocol/drizzle/tree/develop/packages/react-plugin/src/hooks

Reply:
For example, if I naively try to run this hook:

```js
  const call = useCacheCall("Voting", "retrieveRewards");
```

I get this error:

```sh
TypeError: t.contracts[e].methods[n].cacheCall is not a function
```

Reply:
Yeah, makes sense. Drizzle doesn’t handle calling non view functions like view functions, although it should.